src/backend/aspen/app/views/sample.py

```diff
@@ -144,6 +154,95 @@ def _format_lineage(sample: Sample) -> dict[str, Any]:
+            .filter(
+                and_(
+                    Sample.uploaded_pathogen_genome != None,  # noqa: E711
+                    or_(
+                        Sample.private_identifier.in_(sample_ids),
+                        Sample.public_identifier.in_(sample_ids),
+                    ),
+                )
+            )
+            .options(
+                joinedload(Sample.uploaded_pathogen_genome).undefer(
+                    UploadedPathogenGenome.sequence
+                ),
+            )
+        )
+        # check that user has access to the samples
+        group_ids = set(sample.submitting_group_id for sample in all_samples)
+        cansee_groups: Set[int] = {
+            cansee.owner_group_id
+            for cansee in user.group.can_see
+            if cansee.data_type == DataType.SEQUENCES
+        }
+        # add the users own group
+        cansee_groups.add(user.group_id)
+        if not group_ids.issubset(cansee_groups):
+            return Response(
+                "User does not have access the requested sequences",
+                400,
+            )
+
+        cansee_groups_private_identifiers: Set[int] = {
+            cansee.owner_group_id
+            for cansee in user.group.can_see
+            if cansee.data_type == DataType.PRIVATE_IDENTIFIERS
+        }
+        # create output file
+        fasta_file = tempfile.NamedTemporaryFile("w+t")
+        with open(fasta_file.name, "w") as f:
+            for sample in all_samples:
+                if sample.uploaded_pathogen_genome:
+                    pathogen_genome: UploadedPathogenGenome = (
+                        sample.uploaded_pathogen_genome
+                    )
+                    sequence: str = "".join(
+                        [
+                            line
+                            for line in pathogen_genome.sequence.splitlines()  # type: ignore
+                            if not (line.startswith(">") or line.startswith(";"))
+                        ]
+                    )
+                    stripped_sequence: str = sequence.strip("Nn")
+                    # use private id if the user has access to it, else public id
+                    if (
+                        sample.submitting_group_id == user.group_id
+                        or sample.submitting_group_id
+                        in cansee_groups_private_identifiers
+                        or user.system_admin
+                    ):
+                        f.write(f">{sample.private_identifier}\n")  # type: ignore
+                    else:
+                        f.write(f">{sample.public_identifier}\n")
+                    f.write(stripped_sequence)
+                    f.write("\n")
+
+        fasta_filename = f"{user.group.name}_sample_sequences.fasta"
+        res = send_file(
+            fasta_file, attachment_filename=fasta_filename, as_attachment=True
+        )
+        # remove the file
+        f.close()
```

Comment:
interesting, we don't need to close the file before sending it?

Reply:
oh yea the send_file method needs to read the file so it failed if we had already closed the file